### ISSUE TITLE
How to disable AMP for custom post types in a multisite network ?

### ISSUE THREAD

**Issue by edemir206** [Automattic/amp-wp#5782]
Hello,

All of sudden all of our network content are being tretated as AMP pages because something changed on the plugin latest release.

Tried using the filter `amp_supportable_post_types` without any success. How do I use it ?

Also how do I enable AMP for only "POST" post type without going in every website of my 450 website network ?

**Comment by westonruter** [comment]
The `post` post type should be enabled by default. Can you share a screenshot of the Supported Templates settings section for one such site in your network?

**Comment by edemir206** [comment]
Hi @westonruter 

You didn't understand, `post` types are already enabled and working but all my others custom post types are enabled too, so it means everything on my site is AMP enabled now which I do not want. Somehow before I had only `post` as amp enabled for all websites on my network.

What I need is a php code where I can put somewhere (in theme function.php for example) to tell the "amp" plugin to ignore custom post types or to enable only in post types I select, like a settings override (ignore the plugin settings page). I thought `amp_supportable_post_types` does this but I couldn't make it work.

Below is a screenshot of a the settings.

![image](https://user-images.githubusercontent.com/1863099/104844617-0d909680-58b0-11eb-9157-91bd6f412434.png)

**Comment by westonruter** [comment]
Sorry, you're right, I misunderstood. 

You're encountering a bug in the migration logic we had when updating fro, 1.x to 2.x, which we fixed in [v2.0.4](https://github.com/ampproject/amp-wp/releases/tag/2.0.4) via https://github.com/ampproject/amp-wp/pull/5413.

If you want to put some code inside of your theme's `functions.php` to fix this issue you could temporarily add this code which will update the database option to only include the `post` post type:

```php
$supported_post_types = array( 'post' );
if ( class_exists( 'AMP_Options_Manager' ) && $supported_post_types !== AMP_Options_Manager::get_option( 'supported_post_types' ) ) {
	AMP_Options_Manager::update_option( 'supported_post_types', $supported_post_types );
}
```

While this would work, a better solution is to use WP-CLI to do this update across the network. To facilitate that, I've prepared a WP-CLI command extension plugin that adds a new `wp amp update-supported-post-types` command: https://gist.github.com/westonruter/0d12d09119cee62ed2596c9313116c79

Once network-activated, you can update the supported post types to just `post` for one site via:

```bash
wp amp update-supported-post-types post
```

And then to run across all sites on a network you can [combine it](https://danielbachhuber.com/tip/run-wp-cli-command-wordpress-multisite/) with the `wp site list` command:

```bash
wp site list --field=url | xargs -n1 -I % wp --url=% amp update-supported-post-types post
```

You can then deactivate the WP-CLI command extension plugin.
